nibabel/streamlines/trk.py

```diff
@@ -0,0 +1,691 @@
+
+
+def get_affine_trackvis_to_rasmm(header):
+    # TRK's streamlines are in 'voxelmm' space, we will compute the
+    # affine matrix that will bring them back to RAS+ and mm space.
+    affine = np.eye(4)
+
+    # The affine matrix found in the TRK header requires the points to
+    # be in the voxel space.
+    # voxelmm -> voxel
+    scale = np.eye(4)
+    scale[range(3), range(3)] /= header[Field.VOXEL_SIZES]
+    affine = np.dot(scale, affine)
+
+    # TrackVis considers coordinate (0,0,0) to be the corner of the
+    # voxel whereas streamlines returned assume (0,0,0) to be the
+    # center of the voxel. Thus, streamlines are shifted of half a voxel.
+    offset = np.eye(4)
+    offset[:-1, -1] -= 0.5
+    affine = np.dot(offset, affine)
+
+    # If the voxel order implied by the affine does not match the voxel
+    # order in the TRK header, change the orientation.
+    # voxel (header) -> voxel (affine)
+    header_ornt = asstr(header[Field.VOXEL_ORDER])
+    affine_ornt = "".join(aff2axcodes(header[Field.VOXEL_TO_RASMM]))
+    header_ornt = axcodes2ornt(header_ornt)
+    affine_ornt = axcodes2ornt(affine_ornt)
+    ornt = nib.orientations.ornt_transform(header_ornt, affine_ornt)
+    M = nib.orientations.inv_ornt_aff(ornt, header[Field.DIMENSIONS])
+    affine = np.dot(M, affine)
+
+    # Applied the affine found in the TRK header.
+    # voxel -> rasmm
+    voxel_to_rasmm = header[Field.VOXEL_TO_RASMM]
+    affine_voxmm_to_rasmm = np.dot(voxel_to_rasmm, affine)
+    return affine_voxmm_to_rasmm.astype(np.float32)
+
+
+def get_affine_rasmm_to_trackvis(header):
+    return np.linalg.inv(get_affine_trackvis_to_rasmm(header))
+
+
+def encode_value_in_name(value, name, max_name_len=20):
+    """ Encodes a value in the last two bytes of a string.
+
+    If `value` is one, then there is no encoding and the last two bytes
+    are left untouched. This function also verify that the length of name is
+    less than `max_name_len`.
+
+    Parameters
+    ----------
+    value : int
+        Integer value to encode.
+    name : str
+        Name in which the last two bytes will serve to encode `value`.
+    max_name_len : int, optional
+        Maximum length name can have.
+
+    Returns
+    -------
+    encoded_name : str
+        Name containing the encoded value.
+    """
+
+    if len(name) > max_name_len:
+        msg = ("Data information named '{0}' is too long"
+               " (max {1} characters.)").format(name, max_name_len)
+        raise ValueError(msg)
+    elif len(name) > max_name_len - 2 and value > 1:
+        msg = ("Data information named '{0}' is too long (need to be less"
+               " than {1} characters when storing more than one value"
+               " for a given data information."
+               ).format(name, max_name_len - 2)
+        raise ValueError(msg)
+
+    if value > 1:
+        # Use the last two bytes of `name` to store `value`.
+        name = (asbytes(name[:18].ljust(18, '\x00')) + b'\x00' +
```

Comment:
Why the hard-coded 18 here?

I don't understand this function, because I was surprised to see this:

```
In[4]: encode_value_in_name(2, 'foo')
Out[4]: 'foo\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02'
```

Should I have expected that?